nilearn/region_extraction/regionextractor.py

```diff
@@ -0,0 +1,341 @@
+        elif self.threshold_strategy == 'percentile':
+            percentile = 100 - (100 / len_maps) * ratio
+            threshold = scoreatpercentile(
+                np.abs(transformed_maps), percentile)
+            transformed_maps[np.abs(transformed_maps) < threshold] = 0.
+
+        regions_accumulated = []
+        index = []
+
+        for i, trans_map in enumerate(transformed_maps):
+            each_map_data = maps_data[..., i]
+            trans_map_img = self.masker_.inverse_transform(trans_map)
+            trans_map_data = trans_map_img.get_data()
+
+            if self.extractor == 'voxel_wise':
+                label_maps, n_labels = label(trans_map_data)
+                regions = extract_regions(each_map_data, affine,
+                                          label_maps, min_size)
+                regions_accumulated.append(regions)
+            elif self.extractor == 'local_regions':
+                smooth_fwhm = self.smooth_fwhm
+                smooth_data = _smooth_array(trans_map_data,
+                                            affine, fwhm=smooth_fwhm)
+                seeds = peak_local_max(smooth_data, indices=False,
+                                       exclude_border=False)
+                seeds_label, seeds_id = label(seeds)
+                # Assigning "-1" as ignored area to random walker
+                seeds_label[trans_map_data == 0] = -1
+                seeds_map = random_walker(trans_map_data, seeds_label,
+                                          mode='cg_mg')
+                # Again replace "-1" values with "0" for an expected behaviour
+                # to region seperation
+                seeds_map[seeds_map == -1] = 0
+                seeds_label_maps, n_seeds_labels = label(seeds_map)
+                regions = extract_regions(each_map_data, affine,
+                                          seeds_label_maps, min_size)
+                regions_accumulated.append(regions)
+
+        all_regions_imgs = concat_niimgs(regions_accumulated)
+        index.extend([i] * all_regions_imgs.shape[3])
+
+        self.index_ = index
+        self.regions_ = all_regions_imgs
+        return self
+
+    def transform(self, imgs):
+        """ Transform region signals to voxel timeseries signals.
+
+        Parameters
+        ----------
+        imgs: a Niimg-like images/objects
+            Data on which regions signals are transformed to voxel
+            time series signals.
+
+        Returns
+        -------
+        region_signals_: a numpy array
+            a list of averaged timeseries signals of each of the region.
+        """
+        signals = []
+        regions_ = check_niimg_4d(self.regions_)
+        nifti_maps_masker = NiftiMapsMasker(regions_, self.masker_.mask_img_)
+        nifti_maps_masker.fit()
+
+        for img in imgs:
+            region_signals = nifti_maps_masker.transform(img)
+            signals.append(region_signals)
```

Comment:
Well, you probably just want to use the transform method of the Maps Masker, you don't need a `for` loop here.